### ISSUE TITLE
about function loadSystemRoots 

### ISSUE THREAD

**Issue by lysShub** [saferwall/pe#100]
1. `if !strings.Contains(string(out), "command completed successfully") {` not work on other language system；maybe `if cmd.ProcessState.ExitCode() != 0 {`
2. `certutil  -syncWithWU xxx` cost too many time; recommendation use `SystemCertPool` firstly, if verify fail, re-verify use custom pool

**Comment by LordNoteworthy** [comment]
@lysShub thanks for your feedback. 

I agree we should check exit code regarding point number 1.
For point 2: do you know if verify does grab the lastest certs ?

**Comment by lysShub** [comment]
> @lysShub thanks for your feedback.
> 
> I agree we should check exit code regarding point number 1. For point 2: do you know if verify does grab the lastest certs ?

my test result is, when a required cert missing：
    if exec 'certutil verify xxx.crt' with administrator access, the system will automatically download the missing cert
    otherwise, will verify fail

**Comment by lysShub** [comment]
not need to manually create CertPool, if VerifyOptions.Roots is nil and golang will call system API to verify: https://github.com/golang/go/blob/go1.22.0/src/crypto/x509/verify.go#L770

@LordNoteworthy

**Comment by LordNoteworthy** [comment]
Thanks a lot @lysShub, I should be able to fix this by the end of the week

**Comment by LordNoteworthy** [comment]
Hey @lysShub 

`-verify` requires us to first download the certs, so do we have to first run the sync then in subsequent commands do the verification ?

Can you post a quick benchmark how this is faster ?

My tests shows that when `sync` runs the first time, its slow, then subsequent executions are fast.

**Comment by lysShub** [comment]
I mean is that no need the certPool,  but directly：
```go
	// Let's load the system root certs.
	if !pe.opts.DisableCertValidation {
		certValid = pkcs.Verify() == nil
	}
```

**Comment by LordNoteworthy** [comment]
I see, does this work prior to go v1.22 ?

**Comment by lysShub** [comment]
yes, it depend on [Verify](https://pkg.go.dev/crypto/x509@go1.22.0?GOOS=windows#Certificate.Verify)，not need to distinguish Windows or Linux

**Comment by LordNoteworthy** [comment]
ah here: https://github.com/golang/go/blob/go1.22.0/src/crypto/x509/verify.go#L770, let me check that

**Comment by lysShub** [comment]
to be continue, finally call windows system function, such as [syscall.CertGetCertificateChain](https://github.com/golang/go/blob/master/src/crypto/x509/root_windows.go#L250)

**Comment by ayoubfaouzi** [comment]
```go
	// Let's load the system root certs.
	if !pe.opts.DisableCertValidation {
		certValid = pkcs.Verify() == nil
	}
```

The unit tests fails when I changed it as above. Can you please submit a PR ?
